src/LibraryUtilities.ts

```diff
@@ -175,22 +190,52 @@ export function constructLibraryItem(
         // current 'result' as the parent node for child nodes to be appended.
         // 
         let parentNode = inclusive ? null : result;
+        let nodeFound: boolean = false;
+        let isLeafNode: boolean = true;
 
         for (let j = 0; j < typeListNodes.length; j++) {
-
             let fullyQualifiedName = typeListNodes[j].fullyQualifiedName;
-            if (!fullyQualifiedName.startsWith(includePath)) {
-                continue; // Not matching, skip to the next type node.
+            let fullyQualifiedNameParts = fullyQualifiedName.split('.');
+            let lastIncludedPart = includeParts[includeParts.length - 1];
+
+            // Check if a matching node is found, and if the fullyQualifiedName contains
+            // the (whole) last word in includePath. 
+            // E.g. This ensures EllipseArc nodes to be excluded in the Ellipse category.
+            if (fullyQualifiedName.startsWith(includePath) && _.contains(fullyQualifiedNameParts, lastIncludedPart)) {
+
+                // Check if the includePath represents a leaf node.
+                if (includePath.length < fullyQualifiedName.length)
+                    isLeafNode = false;
+
+                parentNode = constructNestedLibraryItems(includeParts,
+                    typeListNodes[j], inclusive, parentNode, layoutElement.include[i].iconUrl);
+
+                if (isLeafNode && parentNode && (parentNode != result)) {
+                    // If the new node created is a leaf node, append it as a child of 
+                    // the current existing node. This step is necessary for nodes that
+                    // have overloads with the same name but different parameters.
+                    result.appendChild(parentNode);
+                    nodeFound = true;
+
+                    // Reset the value of parentNode, to check through the rest of typeListNodes.
+                    parentNode = inclusive ? null : result;
+                }
             }
-
-            parentNode = constructNestedLibraryItems(includeParts,
-                typeListNodes[j], inclusive, parentNode, layoutElement.include[i].iconUrl);
         }
 
-        if (parentNode && (parentNode != result)) {
-            // If a new parent node was created, append it as a child of 
-            // the current resulting node.
-            result.appendChild(parentNode);
+        // If a node that matches the includePath cannot be found
+        if (!nodeFound) { 
+
+            // isLeafNode is set to true if there are no nodes that can match the includePath,
+            // or if the node is a leaf node.
+            if (isLeafNode) {
+                console.warn("Warning: The type '" + includePath + "' is not found in " +
+                    "'RawTypeData.json'. No node of this type is rendered in the library view.");
```

Comment:
`RawTypeData.json` and `LayoutSpecs.json` is just an example JSON file names we happen to use to demo `librarie.js` functionality. In real world when a client consumes `librarie.js`, it does not care about the files at all, the JSON contents will be passed to `librarie.js` in JSON format. This can be simplified as:

```js
console.warn("Matching type not found: " + includePath);
```